infra/main.bicep

```diff
@@ -271,6 +273,46 @@ module searchRoleBackend 'core/security/role.bicep' = {
   }
 }
 
+module searchRoleOpenAi 'core/security/role.bicep' = if (!searchApiKeyEnabled) {
```

Comment:
I see you only did the system roles, what about local server running? For the other repo with these roles, we also have the equivalent for the user. I don't recall if this repo can be run locally or not.

Reply:
So I tried locally and it didn't require me to add any roles. I think this is because we're still calling OpenAI with API key. I missed this before, so will make the change and push to this branch/PR along with the associated required role.

Reply:
We'll want rbac to work for the local user as well.